core/src/main/java/ai/timefold/solver/core/impl/domain/common/accessor/gizmo/GizmoMemberAccessorImplementor.java

```diff
@@ -161,12 +162,17 @@ private static void createConstructor(ClassCreator classCreator, GizmoMemberInfo
                     Method.class, String.class, Class[].class),
                     declaringClass, name,
                     methodCreator.newArray(Class.class, 0));
-            ResultHandle type =
-                    methodCreator.invokeVirtualMethod(
-                            MethodDescriptor.ofMethod(Method.class, "getGenericReturnType", Type.class),
-                            method);
-            methodCreator.writeInstanceField(FieldDescriptor.of(classCreator.getClassName(), GENERIC_TYPE_FIELD, Type.class),
-                    thisObj, type);
+            if (memberInfo.isRequiredReturnType()) {
+                // We create a field to store the result, only if the called method has a return type; otherwise, we
+                // will only execute it
```

Comment:
```suggestion
                // We create a field to store the result, only if the called method has a return type. 
                // Otherwise, we will only execute it.
```

It reads better if the lines are split by sentence, if possible.
(The IntelliJ Grazie Pro plugin helps with writing a lot.)